kornia/color/colormap.py

```diff
@@ -1,126 +1,152 @@
+        }[self.name]
+
+    @classmethod
+    def list(cls):
+        r"""Returns a list of names of enumeration members.
+
+        Returns:
+            List[str]: A list containing the names of enumeration members.
+        """
+        return [c.name for c in cls]
+
+
+class ColorMap:
+    r"""Class to represent a colour map. It can be created or selected from the built-in colour map.
 
-RGBColor = List[float]
-
-
-def _list_color_to_tensor(
-    colors: List[RGBColor], dtype: Optional[torch.dtype], device: Optional[torch.device]
-) -> Tensor:
-    return tensor(list(colors), dtype=dtype, device=device).T
-
+    Args:
+        base: A list of RGB colors to define a new custom colormap or
+        the name of a built-in colormap as str or using CMAP class.
+        num_colors: Number of colors in the colormap.
+        device: The device to put the generated colormap on.
+        dtype: The data type of the generated colormap.
 
-class ColorMap(ABC):
-    r"""Abstract class to represents a color map."""
+    Returns:
+            ColorMap: An object of the colormap with the num_colors length.
 
-    colors: Tensor
+    Example:
+        >>> ColorMap(base='viridis', num_colors=8).colors
+        tensor([[0.2813, 0.2621, 0.2013, 0.1505, 0.1210, 0.2463, 0.5259, 0.8557],
+                [0.0842, 0.2422, 0.3836, 0.5044, 0.6258, 0.7389, 0.8334, 0.8886],
+                [0.4072, 0.5207, 0.5543, 0.5574, 0.5334, 0.4519, 0.2880, 0.0989]])
+
+        # Create color map from first color (RGB with range[0-1]) to last one with num_colors length.
+        >>> ColorMap(base=[[0., 0.5 , 1.0], [1., 0.5, 0.]], num_colors=8).colors
+        tensor([[0.0000, 0.0000, 0.1250, 0.3750, 0.6250, 0.8750, 1.0000, 1.0000],
+                [0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000, 0.5000],
+                [1.0000, 1.0000, 0.8750, 0.6250, 0.3750, 0.1250, 0.0000, 0.0000]])
+    """
 
     def __init__(
         self,
-        base_colormap: List[RGBColor],
-        num_colors: int,
-        device: Optional[torch.device],
-        dtype: Optional[torch.dtype],
+        base: Union[List[RGBColor], str, CMAP],
+        num_colors: Optional[int] = 64,
+        device: Optional[torch.device] = None,
+        dtype: Optional[torch.dtype] = None,
     ) -> None:
+        super().__init__()
         self._dtype = dtype
         self._device = device
 
-        self.colors = interpolate(
-            _list_color_to_tensor(base_colormap, dtype=self._dtype, device=self._device)[None, ...],
-            size=num_colors,
-            mode="linear",
-        )[0, ...]
+        if isinstance(base, str):
+            base = base.lower()
+            if base not in CMAP.list():
+                raise ValueError(f"Unsupported colormap: {base}. Available colormaps are {CMAP.list()}")
+            base_colormap_data = CMAP[base]._load_base()
+            self.name = base
+        elif isinstance(base, CMAP):
+            # Check if CMAP.base exist in enum CMAP? How?
```

Comment:
what do you mean?

Reply:
If we recieve an input "CMAP.test" that it does not exist in our enum, what happen? Do we need to check if it exists?

Reply:
I'm assuming that if it's available in the enum, it will have an associated base. If the value/map doesn't exist in the enum, I don't think it's possible to create it.